compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -3726,8 +3726,8 @@ public void visit(BLangForeach foreach, AnalyzerData data) {
         // Create a new block environment for the foreach node's body.
         SymbolEnv blockEnv = SymbolEnv.createBlockEnv(foreach.body, currentEnv);
         // Check foreach node's variables and set types.
-        handleForeachDefinitionVariables(foreach.variableDefinitionNode, foreach.varType, foreach.isDeclaredWithVar,
-                                         false, blockEnv);
+        handleForeachDefinitionVariables(foreach.variableDefinitionNode, Types.getReferredType(foreach.varType),
```

Comment:
If we do `getReferredType()` at this level; we loose the type reference information here itself. We'll have to handle at the lower levels where we actually need the actual type and use the reference type in the errors even so.
eg:

```ballerina
type Foo record {|
int id;
|};

type Bar Foo;
public function dummyTest() {
    Bar[] arr = [];
    foreach int item in arr { //now error is expected 'Foo' found int. But it should be 'Bar'

    }
}
```